File: client/src/main/java/net/runelite/client/plugins/oneclickagility/PyramidCourse.java
package net.runelite.client.plugins.oneclickagility;

import net.runelite.api.TileObject;

import java.util.List;

public class PyramidCourse extends Course
{
    PyramidCourse(List<Integer> obstacleIDs, List<ObstacleArea> obstacleAreaList)
    {
        super(obstacleIDs, obstacleAreaList);
    }

    @Override
    void addObstacle(TileObject obstacle)
    {
        for (ObstacleArea area:obstacleAreaList)
        {
            if (obstacle.getId() == area.getNextObstacleID() && area.containsObject(obstacle))
            {
                area.setNextObstacle(obstacle);
                return;
            }
        }
    }

    @Override
    void removeObstacle(TileObject obstacle)
    {
        for (ObstacleArea area:obstacleAreaList)
        {
            if (obstacle.getId() == area.getNextObstacleID() && area.containsObject(obstacle))
            {
                area.setNextObstacle(null);
                return;
            }
        }
    }
}

File: client/src/main/kotlin/meteor/plugins/oneclickagility/PyramidCourse.kt
package meteor.plugins.oneclickagility

import net.runelite.api.TileObject

class PyramidCourse internal constructor(obstacleIDs: List<Int>, obstacleAreaList: List<ObstacleArea>) :
    Course(obstacleIDs, obstacleAreaList) {
    override fun addObstacle(obstacle: TileObject) {
        for (area in obstacleAreaList) {
            if (obstacle.id == area.nextObstacleID && area.containsObject(obstacle)) {
                area.nextObstacle = obstacle
                return
            }
        }
    }

    override fun removeObstacle(obstacle: TileObject) {
        for (area in obstacleAreaList) {
            if (obstacle.id == area.nextObstacleID && area.containsObject(obstacle)) {
                area.nextObstacle = null
                return
            }
        }
    }
}